src/Features/Onboarding.php

```diff
@@ -311,14 +311,51 @@ public function component_settings( $settings ) {
 
 		// Only fetch if the onboarding wizard is incomplete.
 		if ( $this->should_show_profiler() ) {
-			$settings['onboarding']['productTypes'] = self::get_allowed_product_types();
-			$settings['onboarding']['themes']       = self::get_themes();
-			$settings['onboarding']['activeTheme']  = get_option( 'stylesheet' );
+			$settings['onboarding']['productTypes']  = self::get_allowed_product_types();
+			$settings['onboarding']['themes']        = self::get_themes();
+			$settings['onboarding']['activeTheme']   = get_option( 'stylesheet' );
+			$settings['onboarding']['activePlugins'] = self::get_active_plugins();
 		}
 
 		return $settings;
 	}
 
+	/**
+	 * Gets an array of plugins that can be installed & activated via the onboarding wizard.
+	 */
+	public static function get_allowed_plugins() {
+		return apply_filters(
+			'woocommerce_onboarding_plugins_whitelist',
+			array(
+				'jetpack'              => 'jetpack/jetpack.php',
+				'woocommerce-services' => 'woocommerce-services/woocommerce-services.php',
+			)
+		);
+	}
+	/**
+	 * Get a list of active plugins, relevent to the onboarding wizard.
+	 *
+	 * @return array
+	 */
+	public static function get_active_plugins() {
+		$all_active_plugins   = get_option( 'active_plugins', array() );
+		$allowed_plugins      = self::get_allowed_plugins();
+		$active_plugin_files  = array_intersect( $all_active_plugins, $allowed_plugins );
+		$allowed_plugin_slugs = array_flip( $allowed_plugins );
+		$active_plugins = array();
+		foreach ( $active_plugin_files as $file ) {
+			$slug = $allowed_plugin_slugs[ $file ];
+			if ( 'jetpack' === $slug || 'woocommerce-services' === $slug ) {
+				// If Jetpack is installed but not connected, consider both WCS and Jetpack inactive for the purpose of onboarding.
```

Comment:
So if Jetpack is not activated, we also consider wc services not activated?  Should we return the activated here so we can determine what needs to be activated?

I'm planning on soon implementing the Jetpack connection wc-api methods to check for an active connection if that makes any difference to the checks here.

Reply:
> So if Jetpack is not activated, we also consider wc services not activated?

For the purposes of the designs and benefits screen, yes. You can technically go to the plugins screen and activate WooCommerce Services without Jetpack active/connected, but it won't do anything.

We could be a bit smarter and detect some of these other cases, but I think that might be over optimizing. I'd say most new users running through onboarding or the task list should never have only WCS active. That seems like something you would only hit as a developer or re-running the wizard, which I'm not too concerned about. Activating or trying to install the plugins again won't damage anything -- the API calls will just return faster.